### PR TITLE
`azurerm_mssql_managed_database` - support for retention policies

### PR DESCRIPTION
**Pull request by catriona-m**

### CODE REVIEW

File: website/docs/r/mssql_managed_database.html.markdown [hunk 1]
```diff
@@ -61,8 +61,21 @@ The following arguments are supported:
 
 * `managed_instance_id` - (Required) The ID of the Azure SQL Managed Instance on which to create this Managed Database. Changing this forces a new resource to be created.
 
+* `long_term_retention_policy` - (Optional) A `long_term_retention_policy` block as defined below.
+
+* `short_term_retention_days` - (Optional) The backup retention period in days. This is how many days Point-in-Time Restore will be supported.
```
Comment (katbyte) [inline website/docs/r/mssql_managed_database.html.markdown:66]:
```suggestion
* `short_term_retention_days` - (Optional) The backup retention period in days. This is how many days Point-in-Time Restore will be supported for.
```

File: website/docs/r/mssql_managed_database.html.markdown [hunk 2]
```diff
@@ -61,8 +61,21 @@ The following arguments are supported:
...
+
 ---
 
+A `long_term_retention_policy` block supports the following:
+
+* `weekly_retention` - (Optional) The weekly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 520 weeks. e.g. `P1Y`, `P1M`, `P1W` or `P7D`.
+* `monthly_retention` - (Optional) The monthly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 120 months. e.g. `P1Y`, `P1M`, `P4W` or `P30D`.
+* `yearly_retention` - (Optional) The yearly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 10 years. e.g. `P1Y`, `P12M`, `P52W` or `P365D`.
```
Comment (katbyte) [inline website/docs/r/mssql_managed_database.html.markdown:74]: should these be
```suggestion
* `weekly_retention_duration` - (Optional) The weekly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 520 weeks. e.g. `P1Y`, `P1M`, `P1W` or `P7D`.
* `monthly_retention_duration` - (Optional) The monthly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 120 months. e.g. `P1Y`, `P1M`, `P4W` or `P30D`.
* `yearly_retention_duration` - (Optional) The yearly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 10 years. e.g. `P1Y`, `P12M`, `P52W` or `P365D`.
```
or even this as retention is in the block name?
```suggestion
* `weekly_duration` - (Optional) The weekly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 520 weeks. e.g. `P1Y`, `P1M`, `P1W` or `P7D`.
* `monthly_duration` - (Optional) The monthly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 120 months. e.g. `P1Y`, `P1M`, `P4W` or `P30D`.
* `yearly_duration` - (Optional) The yearly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 10 years. e.g. `P1Y`, `P12M`, `P52W` or `P365D`.
```

**Review by katbyte — APPROVED** [review]
Aside from 2 comments LGTM 💾

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Looks like there is a default value that is messing up the basic test as well

```
=== CONT  TestAccMsSqlManagedDatabase_basic
testcase.go:110: Step 1/2 error: After applying this test step, the plan was not empty.
stdout:
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
~ update in-place
Terraform will perform the following actions:
# azurerm_mssql_managed_database.test will be updated in-place
~ resource "azurerm_mssql_managed_database" "test" {
id                        = "/subscriptions/*******/resourceGroups/acctestRG1-sql-230308173255406144/providers/Microsoft.Sql/managedInstances/acctestsqlserver230308173255406144/databases/acctest-230308173255406144"
name                      = "acctest-230308173255406144"
- short_term_retention_days = 7 -> null
# (1 unchanged attribute hidden)
# (1 unchanged block hidden)
}
Plan: 0 to add, 1 to change, 0 to destroy.
```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @catriona-m - Spotted a couple things while passing through, I think the tests have highlighted one of the points on the default value for `short_term_retention_days`?

File: website/docs/r/mssql_managed_database.html.markdown (hunk 2 above)
Comment (catriona-m) [reply website/docs/r/mssql_managed_database.html.markdown:74]: I've named them this way to keep them consistent with the equivalent in azurerm_mssql_database: https://github.com/hashicorp/terraform-provider-azurerm/blob/fb309bb4be0846c349888940f8e01f03947aab0a/internal/services/mssql/helper/sql_retention_policies.go#L24

But happy to change the names here if that makes more sense?

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Pull request merged**